testsuite/cluster-test/src/experiments/client_compatibility_test.rs

```diff
@@ -5,65 +5,151 @@
+        let num_validators = context.cluster.validator_instances().len();
+        let config_cmd = format!(
+            "/opt/libra/bin/config-builder faucet -o /opt/libra/etc --chain-id {} -s {} -n {}; echo $?; cat /opt/libra/etc/waypoint.txt",
+            ChainId::test(),
+            CFG_SEED,
+            num_validators
+        );
+        let env_cmd = format!(
+            "CFG_CHAIN_ID={} AC_HOST={} AC_PORT={}",
+            ChainId::test(),
+            self.test_host.ip(),
+            self.test_host.ac_port()
+        );
+        let run_cmd = format!("gunicorn --bind 0.0.0.0:{} --access-logfile - --error-logfile - --log-level debug --pythonpath /opt/libra/bin server", faucet_port);
+        // run config builder, set ac and cfg envs, run faucet
+        let full_faucet_cmd = format!("{}; {} {}", config_cmd, env_cmd, run_cmd);
+        let msg = format!("1. Starting faucet on node {}", self.test_host);
+        info!("{}", msg);
+        context.report.report_text(msg);
+        let faucet_job_name = self
+            .test_host
+            .spawn_job(&test_image, &full_faucet_cmd, "run-faucet")
             .await?;
+        info!(
+            "Job {} started for node {}:{} faucet command: {}",
+            faucet_job_name,
+            self.test_host,
+            self.test_host.peer_name(),
+            full_faucet_cmd
+        );
 
-        // do some other jobs
-        info!("Wait for the other jobs to finish");
+        // wait for faucet to finish starting
+        info!("Waiting for faucet job to spin up completely");
         time::delay_for(Duration::from_secs(20)).await;
 
+        // submit a mint request from the test node to the test host (faucet)
+        let run_cli_cmd = format!("/opt/libra/bin/cli --url {} --chain-id {} -f http://{}:{} --waypoint $(cat /opt/libra/etc/waypoint.txt)", self.test_node.json_rpc_url(), ChainId::test(), self.test_host.ip(), faucet_port);
+        let mut cli_cmd = String::new();
+        let cli_cmd_file = "/opt/libra/etc/cmds.txt";
+        let cmds = include_str!("client_compatibility_cmds.txt");
+        for cmd in cmds.split('\n') {
+            cli_cmd.push_str(&format!("echo {} >> {};", cmd, cli_cmd_file));
+        }
+        // config builder, build cli cmd on a single line to file, pipe the file input to cli
+        let full_cli_cmd = format!(
+            "{}; {} {} < {} && echo SUCCESS",
+            config_cmd, cli_cmd, run_cli_cmd, cli_cmd_file
+        );
+        let msg = format!("2. Running CLI mint from node {}", self.test_node);
+        info!("{}", msg);
+        context.report.report_text(msg);
+        info!(
+            "Job starting for node {}:{} CLI command: {}",
+            self.test_node,
+            self.test_node.peer_name(),
+            full_cli_cmd
+        );
+        // run CLI commands as blocking
+        self.test_node
+            .cmd(&test_image, &full_cli_cmd, "run-cli-commands")
+            .await?;
```

Comment:
In past debugging exercises it was proven to be useful to context on error(here and in other similar places):

```
.await.map_err(|err|format_err!("CLI failed: {}", err))?
```